src/renderer/atlas/AtlasEngine.h

```diff
@@ -175,19 +175,24 @@ namespace Microsoft::Console::Render
                 _data{ allocate(size) },
                 _size{ size }
             {
+                std::uninitialized_default_construct_n(_data, size);
             }
 
             Buffer(const T* data, size_t size) :
                 _data{ allocate(size) },
                 _size{ size }
             {
-                static_assert(std::is_trivially_copyable_v<T>);
-                memcpy(_data, data, size * sizeof(T));
+                // Changing the constructor arguments to accept std::span might
+                // be a good future extension, but not to improve security here.
+                // You can trivially construct std::span's from invalid ranges.
+                // Until then the raw-pointer style is more practical.

```

Comment:
I intended the comment more of an "excuse" why following the C++ guidelines there is impractical and wouldn't help with security either.

I was about to write that I'll add it anyways, but just realized that it would require us to include `<span>` everywhere. 🤔 I'd rather replace `<gsl/span>` with `<span>` everywhere first.